src/libraries/System.Transactions.Local/src/System/Transactions/DtcProxyShim/DtcProxyShimFactory.cs

```diff
@@ -350,7 +352,13 @@ internal ITransactionTransmitter GetCachedTransmitter(ITransaction transaction)
     }
 
     internal void ReturnCachedTransmitter(ITransactionTransmitter transmitter)
-        => _cachedTransmitters.Enqueue(transmitter);
+    {
+        if (_cachedTransmitters.Count < _maxCachedInterfaces)
```

Comment:
`ConcurrentQueue<T>.Count` isn't as cheap as it is on some other collections.  It might be fine, but something to keep in mind.

Reply:
Thanks, didn't know that. I'm guessing that this wouldn't be significant in the larger scheme of things, but would a ConcurrentBag be a better choice here (it's just a cache, ordering isn't important)? I assume I can't use MS.Extensions.ObjectPool.ObjectPool here?

Reply:
ConcurrentQueue is probably fine... I just wanted to call it out in case this was incredibly perf sensitive.